include/kernel/thread.h

```diff
@@ -309,3 +285,28 @@ void awaken(void);
 void postpone(sched_context_t *sc);
 #endif
 
+/* Update the kernels timestamp and stores in ksCurTime.
+ * The difference between the previous kernel timestamp and the one just read
+ * is stored in ksConsumed.
+ *
+ * Should be called on every kernel entry where threads can be billed.
+ * May be called without holding the kernel lock!
```

Comment:
```suggestion
 * Can be called without holding the kernel lock because no global data is written.
```

Reply:
The whole point of the comment was to warn people who may want to change this function to be very careful. Perhaps I should change the "May" into "Will".

Currently global data is written, because although it's per node, it may be read cross-node. Hopefully that will get fixed soon though (#854).

I don't like the "because no global data is written" because that implies that that is the only condition before this is thread-safe, which is not true.

Reply:
ah ok, missed that. Maybe make it even a compile time assert then, to this place it found easily once this changes.

Reply:
I wonder, now that we call `updateTimestamp()` on every kernel entry, would it be worth making the `numDomain` check compiled in conditionally at non-SMP only? Might save a few instructions then.